btcjson/chainsvrresults.go

```diff
@@ -238,6 +238,7 @@ type GetBlockFilterResult struct {
 type GetBlockTemplateResultTx struct {
 	Data    string  `json:"data"`
 	Hash    string  `json:"hash"`
+	TxID    string  `json:"txid"`
```

Comment:
Prefer adding the `omitempty` tag here, otherwise, the response on the rpcserver side will include an empty string. Please also add a comment like this:

```
// TODO: remove omitempty once implemented in rpcserver
```